packages/validators/src/schemas/v2023_09_28/experience.ts

```diff
@@ -0,0 +1,186 @@
+import { z } from 'zod';
+import { SchemaVersions } from '../schemaVersions';
+
+const propertyKeySchema = z.string().regex(/^[a-zA-Z0-9-_.]{1,64}$/);
```

Comment:
Can we restrict this further to what [nanoid](https://github.com/ai/nanoid) uses? I know this is the regex for CF entity sys IDs, but I would rather reserve characters like `.` for now, unless we already generate IDs using it in the app today. A smaller key length should be sufficient as well, considering these IDs serve the purpose of a simple lookup within the same doc.

```suggestion
const propertyKeySchema = z.string().regex(/^[A-Za-z0-9_-]{1,21}$/);
```

Reply:
@DanweDE I am using this schema for both the user specified property names (variables) and the nanoid ids. Limiting to 21 is short for user property names, no? I could define two different Schemas here if we want to be more strict.

I agree about removing the `.`, so I'll remove that.

Reply:
In the end, I added two different regex validations, one for nanoid ids (21 chars) and one for property names (64 chars), but let me know if you have any more comments.